egs2/java/asr1/local/java_data_prep.py

```diff
@@ -0,0 +1,70 @@
+"""
+80% 10% 10% split
+"""
+
+import os
+import random
+
+
+if __name__ == "__main__":
+    tsv_path = "downloads/utt_spk_text.tsv"
+
+    with open(tsv_path, "r") as inf:
+        tsv_lines = inf.readlines()
+    tsv_lines = [l.strip() for l in tsv_lines]
+
+    spk2utt = {}
+    utt2text = {}
+    for l in tsv_lines:
+        l_list = l.split("\t")
+        fid = l_list[0]
+        spk = l_list[1]
+        text = l_list[2]
+        path = "downloads/data/%s/%s.flac" % (fid[:2], fid)
+        if os.path.exists(path):
+            utt2text[fid] = text
+            if spk in spk2utt:
+                spk2utt[spk].append(fid)
+            else:
+                spk2utt[spk] = [fid]
+
+    spks = sorted(list(spk2utt.keys()))
+    random.Random(0).shuffle(spks)
+    num_train = int(len(spks) * 0.8)
```

Comment:
I found this in the original paper for this dataset, maybe it would be better to follow their split? We may have some utterances from training samples as a development set.
```

We followed an identical procedure for all languages: The
full dataset was divided into two subsets, one for training, and
the other for testing. The test data was produced by grouping
the data by SpeakerID and holding the first 2000 lines out;
the remainder was used for training.
```